doc/SONiC_QoS_Mgmt_HLD-1.1.md

```diff
@@ -0,0 +1,649 @@
+    Configuring WRED on Green color only supported in this release.
+
+
+- Scheduler Policy  
+
+	openconfig-qos-elements.yang::qos-scheduler-policy-config
+
+
+- DSCP to TC map  
+  No Open Config YANG available.  
+  Augment Open Config YANG model with new definitions similar to [SONIC YANG for DSCP-to-TC-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-dscp-tc-map.yang)
+
+
+- Dot1p to TC map  
+
+  No Open Config YANG available.  
+  Augment Open Config YANG model with new definitions similar to [SONIC YANG for DOT1P-to-TC-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-dot1p-tc-map.yang)
+
+
+- TC to Queue map  
+
+  No Open Config YANG available.   
+  Augment Open Config YANG model with new definition similar to [SONIC YANG for TC-to-Queue-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-tc-queue-map.yang)
+
+
+
+- TC to Priority group  
+
+  No Open Config YANG available.   
+  Augment Open Config YANG model with new definition similar to [SONIC YANG for TC-to-priority-group-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-tc-priority-group-map.yang)
+
+
+- PFC priority to Queue  
+
+  No Open Config YANG available.   
+  Augment Open Config YANG model with new definition similar to [SONIC YANG for PFC-pirority-to-queue-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-pfc-priority-queue.yang)
+
+
+- PFC Configuration
+
+  No open config support for enable PFC priorities on interface.
+  Augment Open Config YANG model with new attribute for PFC priorities and PFC Asynchronous mode.
+
+
+- Interface QoS  
+
+  openconfig-qos-interfaces.yang:: qos-interfaces-config
+  openconfig-qos-elements.yang:: qos-queue-config
+
+
+- QoS Statistics  
+
+  openconfig-qos-interfaces.yang:: qos-interfaces/interface/interface-id/output/queues/queue/state  
+  Augment Open Config YANG model with new definitions for queue statistics and watermark statistics.
+
+
+- QoS Clear Statistics  
+
+  No Open Config YANG available.   
+  RPC based sonic YANG model with new definitions for clear QOS statistics and watermark statistics.
+
+
+- Threshold Breach  
+
+  Augment Open Config YANG model with new definitions of Thresholds breach get.
+
+
+
+### 3.2.2 CLI
+
+#### 3.2.2.1 Configuration Commands
+
+- Configuring WRED
+````
+  sonic(config)# wred <name>
+  sonic(conf-wred-<name>)# random-detect color [green|yellow|red] minimum-threshold 100 maximum-threshold 300
+  drop-probability 40
+  sonic(conf-wred-<name>)# random-detect ecn <ecn_none/green/all>
+
+  sonic(config)# no wred <name>
+````
+- Configure Scheduler Policy for queues
+````
+  sonic(config)# scheduler-policy <name>
+  sonic(conf-sched-policy)# scheduler queue <q#>

```

Comment:
CLI commands pattern for queue-level policies for Scheduler and WRED are different. We should have consistency.
Current WRED creation and binding to queue is:
WRED:```

sonic(config)# wred wred1
sonic(conf-wred-wred1)#
end            Exit to the exec Mode
exit           Exit from current mode
no             Delete WRED profile
random-detect  WRED configuration

sonic(conf-qos-Ethernet52)# queue 3 wred-profile wred1
```
But Binding scheduler to a queue is different.

Can we make it like below:
Queue Scheduler/Shaper:
`sonic(config)# scheduler-policy name1
sonic(conf-sched-policy-name)#
end        Exit to the exec Mode
exit       Exit from current mode
cir
cbs
pir
pbs
priority   scheduler is set to strict priority
no         Negate a commands
weight     weight of DWRR
sonic(conf-qos-Ethernet52)# queue 3 scheduler-policy  name`

Port Shaper:
```
sonic(config)# scheduler-policy name1
sonic(conf-sched-policy-name)#
end        Exit to the exec Mode
exit       Exit from current mode
cir
cbs
pir
pbs
priority   scheduler is set to strict priority
no         Negate a commands
weight     weight of DWRR
sonic(conf-qos-Ethernet52)# scheduler-policy  name1
```